core/src/main/scala/org/apache/spark/scheduler/MapStatus.scala

```diff
@@ -52,8 +55,41 @@ private[spark] sealed trait MapStatus {
    * partitionId of the task or taskContext.taskAttemptId is used.
    */
   def mapId: Long
+
 }
 
+private[spark] class MapStatusLocationFactory(conf: SparkConf) {
+  private val locationExtension = classOf[Location]
+  private val (locationConstructor, locationName) = {
+    conf.get(config.SHUFFLE_LOCATION_PLUGIN_CLASS).map { className =>
+      val clazz = Utils.classForName(className)
+      require(locationExtension.isAssignableFrom(clazz),
+        s"$className is not a subclass of ${locationExtension.getName}.")
+      (clazz.getConstructor(), className)
+    }.orNull
+  }
+
+  private lazy val locationCache: LoadingCache[Location, Location] = CacheBuilder.newBuilder()
+    .maximumSize(10000)
```

Comment:
1. Is it enough by design?
2. Shall we make a conf for this? Otherwise, we may add that conf after we hit some issue in the future.

Reply:
The number (10000) actually comes from the `blockManagerIdCache`. 

Make a conf sounds good to me.